src/AppBundle/Model/DecklistManager.php

```diff
@@ -226,12 +226,12 @@ public function findDecklistsWithComplexSearch()
 				foreach ($cards_code as $i => $card_code) {
 					/* @var $card \AppBundle\Entity\Card */
 					$card = $this->doctrine->getRepository('AppBundle:Card')->findOneBy(array('code' => $card_code));
-					if ($card->getType()->getCode() == "investigator"){
+					if ($card->getType()->getCode() == "hero" || $card->getType()->getCode() == "alter_ego"){

```

Comment:
So it seems like search for 'hero' is working but not alter-ego, I assume because the alter-ego is the linked card.

I think for alter_ego you need the following code:
```
					if ($card->getType()->getCode() == "alter_ego" && !empty($card->getLinkedFrom())) {
						$qb->innerJoin('d.character', "s$i");
						$qb->andWhere("s$i.code = :card$i");
						$qb->setParameter("card$i", $card->getLinkedFrom()[0]->getCode());
						$packs[] = $card->getPack()->getId();
					}
```